solr/solrj-streaming/src/test/org/apache/solr/client/solrj/io/stream/BadClusterTest.java

```diff
@@ -97,15 +102,15 @@ private void testAllNodesDown() throws Exception {
   }
 
   private void testClusterShutdown() throws Exception {
-
-    CloudSolrStream stream = new CloudSolrStream(buildSearchExpression(), streamFactory);
     cluster.shutdown();
 
-    try {
-      getTuples(stream);
-      fail("Expected IOException: SolrException: TimeoutException");
-    } catch (IOException ioe) {
-      SolrException se = (SolrException) ioe.getCause();
+    try (CloudSolrClient cloudSolrClient =
+        new CloudHttp2SolrClient.Builder(Collections.singletonList(zkHost), Optional.empty())
+            .withZkConnectTimeout(1, TimeUnit.SECONDS)
+            .build()) {
+      cloudSolrClient.getById(collection, "1");
```

Comment:
the point was to use stream API I suppose

Reply:
Ahh I guess so, but it is a slow test because of you can't override the ZkConnectTimeout of the Streaming client.

Reply:
I'll revert. The test will just be slow until https://issues.apache.org/jira/browse/CURATOR-720 is implemented.